services/navigation/builtin/builtin.go

```diff
@@ -229,23 +226,23 @@ func (svc *builtIn) computeCurrentBearing(ctx context.Context, path []*geo.Point
 	return fixAngle(path[pathLen-2].BearingTo(path[pathLen-1])), nil
 }
 
-func (svc *builtIn) startWaypoint(extra map[string]interface{}) error {
+func (svc *builtIn) startWaypoint(cancelCtx context.Context, extra map[string]interface{}) {
 	svc.activeBackgroundWorkers.Add(1)
 	utils.PanicCapturingGo(func() {
 		defer svc.activeBackgroundWorkers.Done()
 
 		path := []*geo.Point{}
 		for {
-			if !utils.SelectContextOrWait(svc.cancelCtx, 500*time.Millisecond) {
+			if !utils.SelectContextOrWait(cancelCtx, 500*time.Millisecond) {
 				return
 			}
-			currentLoc, _, err := svc.movementSensor.Position(svc.cancelCtx, extra)
+			currentLoc, _, err := svc.movementSensor.Position(cancelCtx, extra)
 			if err != nil {
 				svc.logger.Errorw("failed to get gps location", "error", err)
 				continue
 			}
 
-			if len(path) <= 1 || currentLoc.GreatCircleDistance(path[len(path)-1]) > .0001 {
+			if len(path) == 0 || currentLoc.GreatCircleDistance(path[len(path)-1]) > .0001 {
```

Comment:
Why is this changing?

Reply:
I misread the original code - I should not have changed this. Thanks for catching this!